### DISCUSSION TITLE
[4일차-윤경] 이벤트 버블링과 캡쳐링이란?

### DISCUSSION THREAD

**Discussion by skyhanull** [zzangCS/zzangInterview#17] [프론트 면접 질문]
이벤트 버블링과 캡쳐링이란?

**Comment by skyhanull** [comment]
이벤트 버블링

- 이벤트 버블링은 특정 화면 요소에서 이벤트가 발생했을 때 해당 이벤트가 더 상위의 화면 요소들로 전달되어 가는 특성을 의미합니다.

이벤트 캡쳐링

- 이벤트 캡쳐는 이벤트 버블링과 반대 방향으로 진행되는 이벤트 전파 방식입니다.

**Comment by smarfy99** [comment]
이벤트 버블링 
- 이벤트가 발생했을 때, 해당 이벤트의 최상단 요소를 만날 때까지  각각의 핸들러가 동작하는 과정
- focus와 같은 이벤트는 버블링 되지 않지만, 거의 모든 이벤트는 버블링된다.
이벤트 캡쳐링
- 이벤트 버블링이 상위 요소로 전달되는 거라면, 이벤트 캡쳐링은 하위 요소로 전파되는 것이다.

**Reply by smarfy99** [reply]
https://ko.javascript.info/bubbling-and-capturing

**Comment by ez1n** [comment]
**이벤트 버블링**
한 요소에 이벤트가 발생하면 할당된 핸들러가 동작하고 최상단의 부모 요소를 만날 때까지 반복되면서 핸들러가 동작하는 현상

**이벤트 캡처링**
부모 요소에서부터 자식 요소로 이벤트를 전달하는 방식
(이벤트 버블링의 반대 현상)
